packages/bot-skeleton/src/scratch/hooks/workspace.js

```diff
@@ -74,8 +74,10 @@ Blockly.Workspace.prototype.dispatchBlockEventEffects = function(event) {
             wait_event.blockId === event.blockId && (wait_event.type === null || event.type === wait_event.type);
 
         if (is_subscribed_event) {
-            wait_event.promise.resolve();
-            this.wait_events.splice(idx, 1);
+            setTimeout(() => {
+                wait_event.promise.resolve();
+                this.wait_events.splice(idx, 1);
+            }, 100);
```

Comment:
can you add a TODO comment here? Since it's just a temporarily solution.

Reply:
Sure. 👍